cpp/unit/unit.h

```diff
@@ -70,6 +71,41 @@ class Unit : public log::LogSource {
 	 */
 	graphic_set *graphics;
 
+	/**
+	 * constructs a new location for this unit replacing any
+	 * existing location
+	 *
+	 * uses same args as the location constructor
+	 * except the first which will filled automatically
+	 */
+	template<class T, typename ... Arg>
```

Comment:
is this variadic templatestuff really necessary? it seems pretty horrible/evil/making it hard to spot errors